src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -535,18 +508,7 @@ void PosixEnginePollerManager::Run(absl::AnyInvocable<void()> cb) {
   }
 }
 
-void PosixEnginePollerManager::TriggerShutdown() {
-  DCHECK(trigger_shutdown_called_ == false);
-  trigger_shutdown_called_ = true;
-  // If the poller is external, dont try to shut it down. Otherwise
-  // set poller state to PollerState::kShuttingDown.
-  if (poller_state_.exchange(PollerState::kShuttingDown) ==
-      PollerState::kExternal) {
-    poller_ = nullptr;
-    return;
-  }
-  poller_->Kick();
-}
+void PosixEnginePollerManager::TriggerShutdown() { poller_->Kick(); }
```

Comment:
This no longer asserts that `TriggerShutdown` was only called once. Is that a safe change to make?